SWLOR.Game.Server/Conversation/BaseManagementTool.cs

```diff
@@ -250,9 +261,64 @@ public override void DoAction(NWPlayer player, string pageName, int responseID)
                 case "RotatePage":
                     RotateResponses(responseID);
                     break;
+                case "RenamePage":
+                    RenameResponses(responseID);
+                    break;
+                case "ConfirmRenamePage":
+                    HandleConfirmSetNameResponse(responseID);
+                    break;
             }
         }
+        private void RenameResponses(int responseID)
+        {
+            GetPC().SetLocalInt("LISTENING_FOR_DESCRIPTION", 1);
+            switch(responseID)
+            {
+                case 1:
+                    string newDescription = GetPC().GetLocalString("NEW_DESCRIPTION_TO_SET");
 
+                    if (string.IsNullOrWhiteSpace(newDescription))
+                    {
+                        _.FloatingTextStringOnCreature("Type in a new name to the chat bar and then press 'Next'.", GetPC().Object, NWScript.FALSE);
+                        return;
+                    }
+
+                    string header = "Your new name follows. If you need to make a change, click 'Back', type in a new description, and then hit 'Next' again.\n\n";
+                    header += _color.Green("New Description: ") + "\n\n";
+                    header += newDescription;
+                    SetPageHeader("ConfirmRenamePage", header);
+                    ChangePage("ConfirmRenamePage");
+                    break;
+            }
+        }
+        private void HandleConfirmSetNameResponse(int responseID)
+        {
+            switch (responseID)
+            {
+                case 1: // Confirm Description Change
+                    var data = _base.GetPlayerTempData(GetPC());
+                    int buildingTypeID = data.TargetArea.GetLocalInt("BUILDING_TYPE");
+                    Enumeration.BuildingType buildingType = buildingTypeID <= 0 ? Enumeration.BuildingType.Exterior : (Enumeration.BuildingType)buildingTypeID;
+                    data.BuildingType = buildingType;
+                    NWPlayer sender = (_.GetPCSpeaker());
+                    if (buildingType == Enumeration.BuildingType.Apartment)
+                    {
+                        Guid pcBaseID = new Guid(data.TargetArea.GetLocalString("PC_BASE_ID"));
+                        var pcBase = _data.Get<PCBase>(pcBaseID);
+                        pcBase.CustomName = GetPC().GetLocalString("NEW_DESCRIPTION_TO_SET");
```

Comment:
Any time you update a database record (IE: Anything you pull back with _data.Get<>, _data.Single<>, etc. etc. You need to submit a change to the database.

In this case, you need to do _data.SubmitDataChange(pcBase, DatabaseActionType.Update);

Changes to database data is not instant. It gets put into a queue in the background and processed in the order the changes were requested. As long as the server doesn't crash, it's guaranteed to save.